cdap-api/src/main/java/co/cask/cdap/api/lineage/operation/Join.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.lineage.operation;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Represents join operation.
+ */
+public class Join extends Operation {
+  private final Constraint constraint;
+  private final Type joinType;
+  private final List<String> outputs;
```

Comment:
How do you represent where the two sides of the join originate from? Don't you need a left and right origin/input operation? To my understanding, all output fields of each of these operations are input to the join. The output is formed by selecting each of the output fields from the side that has a field of that name. The constraint then is a list of pairs of field names. Each representing a field from left and right side that have to match for the join to apply.

Reply:
added Input class representing left and right.